<!-- -*- mode: XML -*- -->
<!-- start wanji_driver/DriverNodelet in a nodelet manager -->

<launch>

  <arg name="device_ip" default="192.168.2.86" />
  <arg name="frame_id" default="wanji" />
  <arg name="manager" default="$(arg frame_id)_nodelet_manager" />
  <arg name="model" default="16E" />
  <arg name="pcap" default="" />
  <arg name="port" default="3001" />
  <arg name="read_fast" default="false" />
  <arg name="read_once" default="true" />
  <arg name="repeat_delay" default="0.0" />
  <arg name="rpm" default="600" />
  <arg name="data_length" default="1001" />
  <arg name="calibration" default="$(find wanji_pointcloud)/data/Wanji_lidar_16/Wanji16-carside.csv" />
 
  <!-- start nodelet manager -->
  <node pkg="nodelet" type="nodelet" name="$(arg manager)" args="manager" />

  <!-- load driver nodelet into it -->
  <node pkg="nodelet" type="nodelet" name="$(arg manager)_driver"
        args="load wanji_driver/DriverNodelet $(arg manager)" >
    <param name="device_ip" value="$(arg device_ip)" />
    <param name="frame_id" value="$(arg frame_id)"/>
    <param name="model" value="$(arg model)"/>
    <param name="pcap" value="$(arg pcap)"/>
    <param name="port" value="$(arg port)" />
    <param name="read_fast" value="$(arg read_fast)"/>
    <param name="read_once" value="$(arg read_once)"/>
    <param name="repeat_delay" value="$(arg repeat_delay)"/>
    <param name="rpm" value="$(arg rpm)"/>
    <param name="data_length" value="$(arg data_length)"/>
    <param name="calibration" value="$(arg calibration)" />
 </node>    

</launch>
